linetools/spectralline.py

```diff
@@ -0,0 +1,370 @@
+"""
+#;+ 
+#; NAME:
+#; spectralline
+#;    Version 1.0
+#;
+#; PURPOSE:
+#;    Module for SpectralLine class
+#;   23-Jun-2015 by JXP
+#;-
+#;------------------------------------------------------------------------------
+"""
+from __future__ import print_function, absolute_import, division, unicode_literals
+
+import numpy as np
+from abc import ABCMeta, abstractmethod
+
+from astropy import constants as const
+from astropy import units as u
+from astropy.units import Quantity
+
+from linetools.spectra import io as lsio
+from linetools.lists.linelist import LineList
+
+#import xastropy.atomic as xatom
+#from xastropy.stats import basic as xsb
+#from xastropy.xutils import xdebug as xdb
+
+# class SpectralLine(object):
+# class AbsLine(SpectralLine):
+
+# Class for Spectral line
+class SpectralLine(object):
+    """Class for a spectral line.  Emission or absorption 
+
+    Attributes:
+        ltype: str
+          type of line, e.g.  Abs, Emiss
+        wrest: float
+          Rest wavelength of the spectral feature
+    """
+    __metaclass__ = ABCMeta
+
+    # Initialize with wavelength
+    def __init__(self, ltype, wrest, linelist=None):
+        """  Initiator
+
+        Parameters
+        ----------
+        ltype : string
+          Type of Spectral line, (Abs
+        wrest : Quantity
+          Rest wavelength
+        linelist : LineList, optional
+          Class of linelist or str setting LineList
+        """
+
+        # Required
+        self.ltype = ltype
+        if ltype not in ['Abs']:
+            raise ValueError('spec/lines: Not ready for type {:s}'.format(ltype))
+
+        # Init
+        if not isinstance(wrest,Quantity):
+            raise ValueError('Rest wavelength must be a Quantity.')
+        self.wrest = wrest
+
+        # Other
+        self.data = {} # Atomic/Moleculare Data (e.g. f-value, A coefficient, Elow)
+        self.analy = {'spec': None, # Analysis inputs (e.g. spectrum; from .clm file or AbsID)
+            'WVMNX': [0., 0.], # Wavelength interval about the line (observed)
+            'VLIM': [0., 0.]*u.km/u.s, # Velocity limit of line, relative to self.attrib['z']
+            'FLG_ANLY': 1 # Analyze
+            }
```

Comment:
I suggest changing

WVMNX -> wlim
VLIM -> vlim
FLG_ANLY -> do_analysis

These seem more intuitive to me. I don't think there's any need for capitals.